src/main/java/org/opensearch/knn/engine/method/EngineSpecificMethodContext.java

```diff
@@ -0,0 +1,31 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ *
+ * Modifications Copyright OpenSearch Contributors. See
+ * GitHub history for details.
+ */
+
+package org.opensearch.knn.engine.method;
+
+import org.opensearch.knn.index.Parameter;
+
+import java.util.Collections;
+import java.util.Map;
+
+/**
+ * Holds context related to a method for a particular engine
```

Comment:
Currently, this information is stored in a [map](https://github.com/opensearch-project/k-NN/blob/main/src/main/java/org/opensearch/knn/index/util/Faiss.java#L213) in the engines. Is the idea to migrate those maps to this abstraction?